### PR TITLE
Set minSdkVersion to 23 (Android 6) II

### PR DESCRIPTION
**Pull request by timkrueger**
Because of updating the minSdkVersion to 23 it comes to some obscure UI
freezes when for different Fresco functionality. To resolve that Fresco is replaced by Coil.

This is already a long living PR and blocks #2415 from merging. Because of that this will be merged with that follow up issues:

- https://github.com/nextcloud/talk-android/issues/2559
- https://github.com/nextcloud/talk-android/issues/2560 

Please observe during the review the behavior of image loading (avatars, previews, ...).

<details>
<summary>Issues in this PR</summary>
- [x] Set current user in search bar of `ConversationListController` 
  
- [x] Image previews in chats are now bigger then before 
  <table>
    <tr>
      <th colspan="2">ImageView with</th>
    </tr>
    <tr>
      <th>wrap_content</th>
      <th>match_parent</th>
    </tr>
    <tr>
      <td></td>
      <td></td>
    </tr>
  </table>
- [x] Configure Coil caching
- [x] Adopt old button style
    <table>
    <tr>
      <td></td>
      <td></td>
    </tr>
  </table>
- [x] Square avatar images in user/group annotations in chat.
</details>

### CODE REVIEW

**Comment by timkrueger** [comment]
I have problems to set the correct icon for the current user in the search bar of the `ConversationListController`. Actually the icon is always blue:




This is the new way with Coil where I try to load the icon: 

https://github.com/nextcloud/talk-android/blob/2ea3a2882c26fff158623057ef22f2996c319e17/app/src/main/java/com/nextcloud/talk/controllers/ConversationsListController.kt#L213-L263

Has somebody an idea whats going wrong?

//cc @AlvaroBrey, @mahibi

**Review by AndyScherzinger — CHANGES_REQUESTED** [review]
@timkrueger 
* user status rendering broken in conversation list
* inline reply in notification has endless spinner on hitting reply button

**Comment by mahibi** [comment]
after switching to light/dark theme, the avatars in conversation list are gone. (after that: when i enter a conversation and go back again, they are visible again)

**Comment by mahibi** [comment]
> * user status rendering broken in conversation list
same here

> * inline reply in notification has endless spinner on hitting reply button
for me the send button disappears when hitting it, but the notification with edited text stays remains..

**Comment by mahibi** [comment]
when i change the own avatar in the settings, it still displays the old avatar in the app, while on web there is the new avatar.

**Comment by mahibi** [comment]
- [ ] checks with Build.VERSION.SDK_INT < Build.VERSION_CODES.M can be removed

**Comment by timkrueger** [comment]
I have one last issue at this point and I don't get it.

If you switch the theme ( regardless from light to dark, dark to light, in Talk or system settings) the avatars in the conversation list are gone:



If you run the app in debug, Coil will write logs. It looks like the the image loading actions will be canceled in the case of a theme switch:

```
2022-12-05 14:29:02.705 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - https://cloud.example.com/index.php/avatar/tim2/512
2022-12-05 14:29:02.729 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@f9c79c1
2022-12-05 14:29:02.734 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@d4cf954
2022-12-05 14:29:02.773 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@a2e8ed9
2022-12-05 14:29:02.826 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@43d1af1
2022-12-05 14:29:02.843 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@8d00a47
2022-12-05 14:29:02.858 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@9b0e0d
2022-12-05 14:29:02.876 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@9be2fc3
```

In comparison the successful loading (during switching back to the list from a single conversation):

```2022-12-05 14:26:41.907 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY_CACHE) - https://cloud.example.com/index.php/avatar/tim/512
2022-12-05 14:26:41.972 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@2987616
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@9e92b97
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@159c184
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@c45cd6d
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@d9ddda2
```

Would appreciate ideas, suggestions or even a little bit pair programming.

**Comment by AndyScherzinger** [comment]
@timkrueger can you rebase and resolve the conflicts, then I'll check it out.

**Comment by AndyScherzinger** [comment]
@timkrueger is ```app/src/main/java/third/parties/fresco/BetterImageSpan.kt``` still needed?

File: app/src/main/java/com/nextcloud/talk/extensions/ImageViewExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            transformations(CircleCropTransformation())
+            placeholder(R.drawable.account_circle_96dp)
+            listener(onError = { _, result ->
+                Log.w(TAG, "Can't load avatar with URL: $url", result.throwable)
+            })
+        }
+    )
+}
+
+@Deprecated("Use function loadAvatar", level = DeprecationLevel.WARNING)
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/extensions/ImageViewExtensions.kt:118]: only 1 call of this method. can be deleted and use loadAvatar instead?

**Review by mahibi — APPROVED** [review]

File: app/src/main/java/com/nextcloud/talk/extensions/ImageViewExtensions.kt (hunk 1 above)
Comment (timkrueger) [reply app/src/main/java/com/nextcloud/talk/extensions/ImageViewExtensions.kt:118]: Would leave it in now. @AlvaroBrey  and I will look together next week how we get it all to android-commons.

- https://github.com/nextcloud/talk-android/issues/2559

**Comment by timkrueger** [comment]
> @timkrueger is `app/src/main/java/third/parties/fresco/BetterImageSpan.kt` still needed?

Still needed at this point. Did not want to tackle the topic as well. Everything was already big enough. But in the future, the part should disappear.

**Pull request merged**